systemtest/src/main/java/io/strimzi/systemtest/performance/utils/TopicOperatorPerformanceUtils.java

```diff
@@ -0,0 +1,295 @@
+     * +-----------------------------------------------------------------------------+
+     * | +-------------+       +-------------+             +-------------+           |
+     * | | Topic 1     |       | Topic 2     |     ...     | Topic N     |           |
+     * | | +---------+ |       | +---------+ |             | +---------+ |           |
+     * | | | Thread  | |       | | Thread  | |             | | Thread  | |           |
+     * | | |         | |       | |         | |             | |         | |           |
+     * | | | Creation| |       | | Creation| |             | | Creation| |           |
+     * | | | Update  | |       | | Update  | |             | | Update  | |           |
+     * | | | Deletion| |       | | Deletion| |             | | Deletion| |           |
+     * | | +---------+ |       | +---------+ |             | +---------+ |           |
+     * | +-------------+       +-------------+             +-------------+           |
+     * +-----------------------------------------------------------------------------+
+     *
+     * @param testStorage           An instance of TestStorage containing configuration and state needed for topic operations.
+     * @param numberOfTopics        The number of Kafka topics to be processed.
+     * @param spareEvents           The number of spare events to be consumed during the process.
+     * @param warmUpTasksToProcess  The number of tasks to warm-up performance and optimize JIT. This number is used just for offsetting.
+     *
+     * @return                      The total time taken to complete all topic lifecycles in milliseconds.
+     */
+    public static long processAllTopicsConcurrently(TestStorage testStorage, int numberOfTopics, int spareEvents, int warmUpTasksToProcess) {
+        final int availableCPUs = Math.max(1, Runtime.getRuntime().availableProcessors());
+        final ExecutorService executor = Executors.newFixedThreadPool(availableCPUs);
+
+        List<CompletableFuture<Void>> futures = new ArrayList<>();
+        ExtensionContext extensionContext = ResourceManager.getTestContext();
+
+        long startTime = System.nanoTime();
+
+        for (int topicIndex = warmUpTasksToProcess; topicIndex < numberOfTopics + warmUpTasksToProcess; topicIndex++) {
+            final int finalTopicIndex = topicIndex;
+            CompletableFuture<Void> future = CompletableFuture.runAsync(() -> performFullLifecycle(finalTopicIndex, testStorage, extensionContext), executor);
+            futures.add(future);
+        }
+
+        // consume spare events
+        for (int j = 0; j < spareEvents; j++) {
+            futures.add(j, CompletableFuture.completedFuture(null));
+        }
+
+        // Wait for all topics to complete their lifecycle
+        CompletableFuture.allOf(futures.toArray(new CompletableFuture[0])).join();
+        LOGGER.info("All topic lifecycles completed.");
+
+        long allTasksTimeMs = (System.nanoTime() - startTime) / 1_000_000;
```

Comment:
It would be good to add some sleep time (e.g. 30s) as cooldown after this line (i.e after each batch completes).

Reply:
Why it's needed?

Reply:
It's just a way to set a boundary between tests, so that one test run is less likely to influence results of the next.

Reply:
Cool I have added
```java
if (warmUpTasksToProcess != 0) {
            // boundary between tests => less likelihood that tests would influence each other
            LOGGER.info("Cooling down");
            Thread.sleep(30_000);
        }
```